include/miniocpp/client.h

```diff
@@ -12,9 +12,11 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
 
-#ifndef _MINIO_S3_CLIENT_H
-#define _MINIO_S3_CLIENT_H
+#ifndef MINIO_CPP_CLIENT_H_INCLUDED
```

Comment:
We would need to define the name include namespace something like. `_MINIO_S3_CLIENT_H_INCLUDED`. This way we may have `_MINIO_ADMIN_CLIENT_H_INCLUDED` in future.

Reply:
We can change the guards at any time, now they just follow the project name and include file name.

I intentionally removed the underscore as everything that starts with underscore and an uppercase character is reserved by the implementation, thus user-code should never use that.

I would always prefer if we can start guards with project name, thus MINIO_CPP or MINIOCPP (that would match the directory) and then I don't really care what follows, but it should be consistent with names of files and not what is inside the files.

Reply:
I guess this is the time we would need to fix these guards. Changing later in time would increase lot of diffs. My point is we would need to accommodate `admin` client as well. Having `MINIO_CPP_CLIENT_H_INCLUDED` is insufficient for that.

Reply:
If it is OK, I can send a PR to https://github.com/minio/minio-cpp/tree/include_hygiene with the fix

Reply:
Yeah, I get that. I think minio-cpp can have nested namespaces, but I think we should name include guards based on the file they are guarding - so if a file is named "miniocpp/file.h" then the guard is `MINIO_CPP_FILE_H_INCLUDED`, which matches the file.

I'm not fan of inventing guard names based on the content of the files they guard, because that's not the purpose of the guard.

Reply:
I'm not sure there is a rule in the static analysis tool to make sure guard names represent file names, but I have seen codebases that had this scripted so when you rename a file, for example, the include guard is updated as well.